### PR TITLE
Add an example of how to attach labels to nodes and use nodeSelectors so...

### PR DESCRIPTION
**Pull request by a-robinson**
... that pods can be scheduled on specific nodes.

### CODE REVIEW

**Comment by a-robinson** [comment]
Issue #2406

File: examples/node-selection/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+## Node selection example
```
Comment (ddysher) [inline examples/node-selection/README.md]: We'd also mention that pinging pod to specific machine using this approach is discouraged.  This is a scheduling problem, not something user needs to take care of.   The plan is to add more resource type for node, and more resource requirement for pod.

File: examples/node-selection/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+This example shows how to assign a pod to a specific node or to one of a set of nodes using node labels and the NodeSelector API resource. You may want to do this if some subset of your machines is special in some way, such as having SSDs attached to them.
+
+### Step Zero: Prerequisites
+
+This example assumes that you have a basic understanding of kubernetes pods and that you have [turned up a Kubernetes cluster](https://github.com/GoogleCloudPlatform/kubernetes#documentation):
+
+### Step One: Attach label to the node
+
+Run `kubectl get nodes` to get the names of the nodes. Pick out the one that you want to add a label to. Note that label keys are not allowed to contain any upper-case letters. Then run `kubectl get node <node-name> -o yaml > node.yaml`. The contents of the file should look something like this:
```
Comment (ddysher) [inline examples/node-selection/README.md]: Strictly, label keys should be DNS label, not just no upper-case letters.

File: examples/node-selection/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+## Node selection example
+
+This example shows how to assign a pod to a specific node or to one of a set of nodes using node labels and the NodeSelector API resource. You may want to do this if some subset of your machines is special in some way, such as having SSDs attached to them.
+
+### Step Zero: Prerequisites
+
+This example assumes that you have a basic understanding of kubernetes pods and that you have [turned up a Kubernetes cluster](https://github.com/GoogleCloudPlatform/kubernetes#documentation):
+
```
Comment (ddysher) [inline examples/node-selection/README.md]: s/:/.

File: examples/node-selection/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+## Node selection example
+
+This example shows how to assign a pod to a specific node or to one of a set of nodes using node labels and the NodeSelector API resource. You may want to do this if some subset of your machines is special in some way, such as having SSDs attached to them.
```
Comment (ddysher) [inline examples/node-selection/README.md]: IIRC, NodeSelector is not an API resource, it's an attribute of Pod spec :)

File: examples/node-selection/README.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  conditions:
+  - kind: Ready
+    lastTransitionTime: null
+    status: Full
+uid: 526a4156-ab42-11e4-9817-42010af0258d
+</pre>
+
+Then update the node by running `kubectl update -f node.yaml`. Make sure that the resourceVersion you use in your update call is the same as the resourceVersion returned by the get call. If something about the node changes between your get and your update, the update will fail because the resourceVersion will have changed.
+
+Note that as of 2015-02-03 there are a couple open issues that prevent this from working without modification. Due to [issue #3005](https://github.com/GoogleCloudPlatform/kubernetes/issues/3005), you have to remove all status-related fields from the file, which is both everything under the `status` field as well as the `hostIP` field. Due to [issue 4041](https://github.com/GoogleCloudPlatform/kubernetes/issues/4041), you may have to modify the representation of the resource capacity numbers to make them integers. These are both temporary, and fixes are being worked on. In the meantime, you would actually call `kubectl update -f node.yaml` with a file that looks like this:
```
Comment (ddysher) [inline examples/node-selection/README.md]: Removing `hostIP` is only required for v1beta1/2.

File: examples/node-selection/README.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      - image: nginx
+        name: nginx
+    id: nginx
+    version: v1beta1
+id: nginx
+kind: Pod
+labels:
+  env: test
+<b>nodeSelector:
+  diskType: ssd</b>
```
Comment (ddysher) [inline examples/node-selection/README.md]: s/diskType/disktype?

**Comment by a-robinson** [comment]
Thanks for the review! I've fixed up everything you mentioned.

**Comment by brendandburns** [comment]
LGTM, personally, I would soften the language around "discouraged" to be something like "generally this is unnecessary, as the scheduler will take care of it for you, but in certain circumstances like ..."

**Comment by a-robinson** [comment]
Sounds good, done.

File: examples/node-selection/README.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+This example shows how to assign a pod to a specific node or to one of a set of nodes using node labels and the nodeSelector field in a pod specification. Generally this is unnecessary, as the scheduler will take care of things for you, but you may want to do so in certain circumstances like to ensure that your pod ends up on a machine with an SSD attached to it.
+
+### Step Zero: Prerequisites
+
+This example assumes that you have a basic understanding of kubernetes pods and that you have [turned up a Kubernetes cluster](https://github.com/GoogleCloudPlatform/kubernetes#documentation).
+
+### Step One: Attach label to the node
+
+Run `kubectl get nodes` to get the names of the nodes. Pick out the one that you want to add a label to. Note that label keys must be in the form of DNS labels (as described in the [identifiers doc](/docs/design/identifiers.md), meaning that they are not allowed to contain any upper-case letters. Then run `kubectl get node <node-name> -o yaml > node.yaml`. The contents of the file should look something like this:
```
Comment (ddysher) [inline examples/node-selection/README.md]: closing paraen

**Comment by ddysher** [comment]
LGTM  one minor comment

File: examples/node-selection/README.md (hunk 7 above)
Comment (a-robinson) [inline examples/node-selection/README.md]: doh! fixed

**Pull request merged**